CRM/Case/BAO/CaseType.php

```diff
@@ -438,20 +434,41 @@ public static function retrieve(&$params, &$defaults) {
   /**
    * @param int $caseTypeId
    *
+   * @deprecated
    * @throws CRM_Core_Exception
-   * @return mixed
+   * @return CRM_Case_DAO_CaseType
    */
   public static function del($caseTypeId) {
-    $caseType = new CRM_Case_DAO_CaseType();
-    $caseType->id = $caseTypeId;
-    $refCounts = $caseType->getReferenceCounts();
-    $total = array_sum(CRM_Utils_Array::collect('count', $refCounts));
-    if ($total) {
-      throw new CRM_Core_Exception(ts("You can not delete this case type -- it is assigned to %1 existing case record(s). If you do not want this case type to be used going forward, consider disabling it instead.", [1 => $total]));
+    return static::deleteRecord(['id' => $caseTypeId]);
+  }
+
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   * @throws CRM_Core_Exception
+   */
+  public static function on_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    // Before deleting a caseType, check references
+    if ($event->entity === 'CaseType' && $event->action === 'delete') {
```

Comment:
If the BAO doesn't listen for foreign entities (ie it only listens for changes on its own data-type), then `self_hook_civicrm_pre` should be better.

Reply:
Thanks, good catch. Updated.